dvc/repo/__init__.py

```diff
@@ -153,22 +150,29 @@ def tree(self, tree):
     def __repr__(self):
         return f"{self.__class__.__name__}: '{self.root_dir}'"
 
+    @cached_property
+    def repo_tree(self):
+        return RepoTree(self.tree, [self], stream=True)
```

Comment:
We need granular configs in each of the tree operations. I could get around with `stream_open()`  or just this ^ hack.
But, we need `fetch` and `stream` to be per-ops rather than per instance.